src/models/service.ts

```diff
@@ -246,4 +252,42 @@ export class ServiceModel extends CosmosdbModelVersioned<
   ): TaskEither<CosmosErrors, Option<RetrievedService>> {
     return super.findLastVersionByModelId([serviceId]);
   }
+
+  public listLastVersionServices(): TaskEither<
+    CosmosErrors,
+    Option<ReadonlyArray<RetrievedService>>
+  > {
+    return tryCatch<CosmosErrors, FeedResponse<RetrievedService>>(
+      () => this.container.items.query(`SELECT * FROM c`).fetchAll(),
+      toCosmosErrorResponse
+    )
+      .map(_ => fromNullable(_.resources))
+      .chain(_ =>
+        _.isSome()
+          ? _.value.length > 0
```

Comment:
We can `AND` those conditions and lower the complexity of the procedure

Reply:
Fixed in 9469f4e